go/teams/seitan_v2.go

```diff
@@ -0,0 +1,224 @@
+		return ikey, fmt.Errorf("invalid token length: expected %d characters, got %d", SeitanEncodedIKeyLength, len(token))
+	}
+	if token[seitanEncodedIKeyV2PlusOffset] != '+' {
+		return ikey, fmt.Errorf("invalid token format: expected %dth character to be '+'", seitanEncodedIKeyV2PlusOffset+1)
+	}
+
+	return SeitanIKeyV2(strings.ToLower(token)), nil
+}
+
+func (ikey SeitanIKeyV2) String() string {
+	return strings.ToLower(string(ikey))
+}
+
+// "Stretched Invite Key"
+type SeitanSIKeyV2 [SeitanScryptKeylen]byte
+
+func (ikey SeitanIKeyV2) GenerateSIKey() (sikey SeitanSIKeyV2, err error) {
+	buf, err := generateSIKey(ikey.String())
+	if err != nil {
+		return sikey, err
+	}
+	copy(sikey[:], buf)
+	return sikey, nil
+}
+
+func (sikey SeitanSIKeyV2) GenerateTeamInviteID() (id SCTeamInviteID, err error) {
+	type InviteStagePayload struct {
+		Stage   string        `codec:"stage" json:"stage"`
+		Version SeitanVersion `codec:"version" json:"version"`
+	}
+
+	payload, err := libkb.MsgpackEncode(InviteStagePayload{
+		Stage:   "invite_id",
+		Version: SeitanVersion2,
+	})
+	if err != nil {
+		return id, err
+	}
+	return generateTeamInviteID(sikey[:], payload)
+}
+
+func (sikey SeitanSIKeyV2) generateKeyPair() (key libkb.NaclSigningKeyPair, err error) {
+	type PrivateKeySeedPayload struct {
+		Stage   string        `codec:"stage" json:"stage"`
+		Version SeitanVersion `codec:"version" json:"version"`
+	}
+
+	payload, err := libkb.MsgpackEncode(PrivateKeySeedPayload{
+		Stage:   "eddsa",
+		Version: SeitanVersion2,
+	})
+	if err != nil {
+		return key, err
+	}
+
+	mac := hmac.New(sha512.New, sikey[:])
+	_, err = mac.Write(payload)
+	if err != nil {
+		return key, err
+	}
+
+	seed := mac.Sum(nil)
+	seed = seed[0:32]
+	pub, priv, err := ed25519.GenerateKey(bytes.NewBuffer(seed))
+	if err != nil {
+		return key, err
+	}
+
+	copy(key.Public[:], pub[:])
+	key.Private = &libkb.NaclSigningKeyPrivate{}
+	copy(key.Private[:], priv[:])
+	return key, nil
+}
+
+func (sikey SeitanSIKeyV2) generatePackedEncryptedKeyWithSecretKey(secretKey keybase1.Bytes32, gen keybase1.PerTeamKeyGeneration, nonce keybase1.BoxNonce, label keybase1.SeitanKeyLabel) (pkey SeitanPKey, encoded string, err error) {
+
+	keyPair, err := sikey.generateKeyPair()
+	if err != nil {
+		return pkey, encoded, err
+	}
+
+	var keyAndLabel keybase1.SeitanKeyAndLabelVersion2
+	keyAndLabel.K = keybase1.SeitanPubKey(keyPair.GetKID().String())
+	keyAndLabel.L = label
+
+	packedKeyAndLabel, err := libkb.MsgpackEncode(keybase1.NewSeitanKeyAndLabelWithV2(keyAndLabel))
+	if err != nil {
+		return pkey, encoded, err
+	}
+	return packAndEncryptKeyWithSecretKey(secretKey, gen, nonce, packedKeyAndLabel, SeitanVersion2)
+}
+
+func (sikey SeitanSIKeyV2) GeneratePackedEncryptedKey(ctx context.Context, team *Team, label keybase1.SeitanKeyLabel) (pkey SeitanPKey, encoded string, err error) {
+	appKey, err := team.SeitanInviteTokenKeyLatest(ctx)
+	if err != nil {
+		return pkey, encoded, err
```

Comment:
i think we should leave as is for consistency since most of the code throughout v1 and v2 uses named returns, unless you think we should refactor throughout

Reply:
Yeah no need to do a big refactor just for this. I do like explicitly returning `nil` though, rather than returning a variable that's "definitely supposed to be nil".